packages/palette/src/elements/Carousel/__tests__/utils.test.ts

```diff
@@ -0,0 +1,135 @@
+import { chunk, compound, paginate } from "../utils"
+
+describe("compound", () => {
+  it("sums correctly (1)", () => {
+    expect(compound([1, 1, 1, 1])).toStrictEqual([1, 2, 3, 4])
+  })
+
+  it("sums correctly (2)", () => {
+    expect(compound([1, 2, 3, 4])).toStrictEqual([1, 3, 6, 10])
+  })
+
+  it("sums correctly (3)", () => {
+    expect(compound([-1, 2, -3, 4])).toStrictEqual([-1, 1, -2, 2])
+  })
+})
+
+describe("chunk", () => {
+  it("chunks correctly (1)", () => {
+    expect(chunk(1000, [500, 500, 500])).toStrictEqual([1000, 500])
+  })
+
+  it("chunks correctly (2)", () => {
+    expect(chunk(999, [500, 500, 500])).toStrictEqual([500, 500, 500])
+  })
+
+  it("chunks correctly (3)", () => {
+    expect(chunk(500, [500, 500, 500])).toStrictEqual([500, 500, 500])
+  })
+
+  it("chunks correctly (4)", () => {
+    expect(chunk(1000, [1000, 500, 500])).toStrictEqual([1000, 1000])
+  })
+
+  it("chunks correctly (5)", () => {
+    expect(chunk(333, [100, 200, 300, 100, 200, 300])).toStrictEqual([
+      300,
+      300,
+      300,
+      300,
+    ])
+  })
+
+  it("chunks correctly (6)", () => {
+    expect(chunk(100, [100, 200, 300, 100, 200, 300])).toStrictEqual([
```

Comment:
Can we use named arguments for these functions? At a glance, its not clear what 100 is, or what the array is for where if this accepted an object things would be a) no longer order dependent; and b) more readable friendly through time.